version.go

```diff
@@ -19,6 +19,10 @@ var (
 	ErrInvalidSemVer = errors.New("Invalid Semantic Version")
 )
 
+// Controls whether or not parsed constraints are cached
+var cacheVersions = true
+var versionCache = make(map[string]*Version)
```

Comment:
This is not a concurrency safe operation. Two concurrent operations can't operate on a map at the same time. Instead you need something with a RWMutex on it to make sure nothing tries to read while it's being updated.